### PR TITLE
Add 'smurl' for CLI API access with certs.

### PR DESCRIPTION
**Pull request by alikins**
This is used for making requests to the RHSM REST
API ("candlepin") from the cli.

It uses the systems rhsm configuration, and uses
identity certificate based authentication by
default.

Bash tab completion is included for url method
completion.

A simple alias or macro form is support for url
methods, for referencing the systems consumer uuid,
or user or org. For example, to show the JSON
result of a GET request (the default) to the
current consumer systems compliance info:

  smurl api --method /consumers/{c}/compliance

-X and -d options are support for specifying the
http request type, and any data to use in PUT or
POST request. This commands mirror curl's options.

### CODE REVIEW

**Comment by dgoodwin** [comment]
How would you feel about dropping the tab completion? While cool, it's probably just something to get out of date and cause problems, and in a place that's not something we can reliably keep up to date. (deployed clients) We really don't need another definition of all API calls to maintain. It's also going to vary depending on the server it's registered to. This is a very technical utility, I'd think it's probably ok to just let people fill out the API calls to make using some other source of data.

**Comment by dgoodwin** [comment]
Also it is a subtle form of zsh user discrimination.

llama

**Comment by dgoodwin** [comment]
Tried to install it but:

RPM build errors:
    Explicit %attr() mode not applicaple to symlink: /tmp/tito/rpmbuild-subscription-managerG4POJO/BUILDROOT/subscription-manager-1.14.0-1.git.13.af7c86b.fc21.x86_64/usr/bin/subscription-manager
    File not found by glob: /tmp/tito/rpmbuild-subscription-managerG4POJO/BUILDROOT/subscription-manager-1.14.0-1.git.13.af7c86b.fc21.x86_64/usr/share/man/man8/smurl.8*

File: man/smurl.8 [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+.fi
+.RE
+
+.PP
+Query the API for the current 'release' of the consumer. Note the use of the {c} alias
+to refer to the systems consumer uuid.
+.RS
+.nf
+[root@server ~]# sudo smurl api --method /consumers/{c}/release
```
Comment (dgoodwin) [inline man/smurl.8]: Love the {c} stuff.

**Comment by alikins** [comment]
re tab completion...

I think I'll see if I can get it to check for a per user file defining the known urls, something like:

```
~/.smurl/candlepin-api-completions-GET
~/.smurl/candlepin-api-completions-POST
``` 

etc. That format picked just because it's what the scripts/gen-api-completions.py generates from
the candlepin_methods.json  (and that was just based around kcoar's ruby version).

I'd try to do it dynamically, but that will be pretty slow.

Maybe stash the completions away in /usr/share/doc/subscription-manager-* or somewhere "unsupported".

**Comment by alikins** [comment]
Hmm. Suppose 'smurl' could try to generate it's completions itself (or via a helper script), that might
make it more Zed'able.

Mostly went with the hardcoded in the script/env just for speed, but hitting a helper isn't uncommon for completions. Should be reasonable as well if it's just reading static data.

**Comment by alikins** [comment]
added man page to Makefile, commit squashing, rebasing

**Pull request closed without merging**